src/bentoml/_internal/cloud/deployment.py

```diff
@@ -490,132 +504,139 @@ def wait_until_ready(
+                stop_tail_thread()
+                sys.exit(1)
 
-            def stop_spinner():
-                spinner.spinner_progress.stop_task(spinner_task_id)
+            signal.signal(signal.SIGINT, signal_handler)
+            console = Console()
 
-            def tail_image_builder_logs():
+            def tail_image_builder_logs(stop_spinner: t.Callable[[], None]):
                 nonlocal ws_session
-                if console is None:
-                    return
 
                 cloud_rest_client = get_rest_api_client(self._context)
+                pod: KubePodSchema | None = None
                 while True:
-                    with lock:
-                        if stop_tail:
-                            return
                     pod = cloud_rest_client.v2.get_deployment_image_builder_pod(
                         self.name, self.cluster
                     )
                     if pod is None:
-                        time.sleep(check_interval)
-                        continue
-                    if pod.pod_status.status != "Running":
-                        time.sleep(check_interval)
+                        if stop_tail_event.wait(check_interval):
+                            return
                         continue
-                    for line, ws in cloud_rest_client.v2.tail_logs(
-                        cluster_name=self.cluster,
-                        namespace=self._schema.kube_namespace,
-                        pod_name=pod.name,
-                        container_name="builder",
-                    ):
-                        with lock:
-                            ws_session = ws
-                        console.print(line, markup=False)
-                    with lock:
-                        ws_session = None
+                    if pod.pod_status.status == "Running":
+                        break
+                    if stop_tail_event.wait(check_interval):
+                        return
+                if pod is None:
+                    console.print(
+                        "🚨 [bold red]Can't found the image builder pod.[/bold red]"
+                    )
                     return
 
+                is_first = True
+                for piece, ws in cloud_rest_client.v2.tail_logs(
+                    cluster_name=self.cluster,
+                    namespace=self._schema.kube_namespace,
+                    pod_name=pod.name,
+                    container_name="builder",
+                    stop_event=stop_tail_event,
+                ):
+                    with lock:
+                        ws_session = ws
+                    decoded_bytes = base64.b64decode(piece)
+                    decoded_str = decoded_bytes.decode("utf-8")
+                    if is_first:
+                        is_first = False
+                        stop_spinner()
+                        console.print("🚧 Image building...")
+                    print(decoded_str, end="")
```

Comment:
@yetone is this expected?

Reply:
@aarnphm Yes, it meets expectations. console.print will be messy (for example, it will do some ANSI color things). What is needed here is direct output to stdout as-is.